dace/transformation/interstate/loop_lifting.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2019-2024 ETH Zurich and the DaCe authors. All rights reserved.
+
+from dace import properties
+from dace.sdfg.sdfg import SDFG, InterstateEdge
+from dace.sdfg.state import ControlFlowRegion, LoopRegion
+from dace.transformation import transformation
+from dace.transformation.interstate.loop_detection import DetectLoop
+
+
+@properties.make_properties
+@transformation.experimental_cfg_block_compatible
+class LoopLifting(DetectLoop, transformation.MultiStateTransformation):
+
+    def can_be_applied(self, graph: transformation.ControlFlowRegion, expr_index: int, sdfg: transformation.SDFG,
+                       permissive: bool = False) -> bool:
+        # Check loop detection with permissive = True, which allows loops where no iteration variable could be detected.
+        # We want this to detect while loops.
+        if not super().can_be_applied(graph, expr_index, sdfg, permissive=True):
+            return False
+
+        # Check that there's a condition edge, that's the only requirement to lift it into loop.
+        cond_edge = self.loop_condition_edge()
+        if not cond_edge or cond_edge.data.condition is None:
+            return False
+        return True
+
+    def apply(self, graph: ControlFlowRegion, sdfg: SDFG):
+        first_state = self.loop_guard if self.expr_index <= 1 else self.loop_begin
+        after = self.exit_state
+
+        loop_info = self.loop_information()
+
+        body = self.loop_body()
+        meta = self.loop_meta_states()
+        full_body = set(body)
+        full_body.update(meta)
+        cond_edge = self.loop_condition_edge()
+        incr_edge = self.loop_increment_edge()
+        inverted = self.expr_index in (2, 3, 5, 6, 7)
```

Comment:
I would prefer that we add `self.loop_is_inverted()` into `DetectLoop`, the reason being that we should try to remove expr_index use from subclass transformations as much as possible (to make it easy to add new patterns). I tried to do this with looptomap etc.

Reply:
Added a `DetectLoop.is_inverted` utility function.